### PR TITLE
Di 18514

### PR DESCRIPTION
**Pull request by michaelatias**

### CODE REVIEW

File: client-side/src/app/components/atd-config/atd-config.component.html [hunk 1]
```diff
@@ -45,12 +45,12 @@
                 </div>
                 <div>
                     <pep-select #pepSelect [value]="Configuration?.CaseQuantityType"
-                        [options]="Actions" [disabled]="Configuration?.InventoryFieldID == '' || Configuration?.ItemConfigFieldID == ''" [emptyOption]='false' [label]="'Uom_AtdConfig_CaseActionTitle' | translate" 
+                        [options]="Actions" [disabled]=" '' || Configuration?.ItemConfigFieldID == ''" [emptyOption]='false' [label]="'Uom_AtdConfig_CaseActionTitle' | translate"
```
Comment (Chasky-H) [inline client-side/src/app/components/atd-config/atd-config.component.html:48]: `'' ||` ?

File: client-side/src/app/components/atd-config/atd-config.component.ts [hunk 2]
```diff
@@ -134,8 +133,10 @@ export class AtdConfigComponent implements OnInit {
             case 'ItemConfig': {
                 this.Configuration.ItemConfigFieldID = $event;
                 if($event == '') {
+                    this,this.Configuration.InventoryType = 'DoNothing'
```
Comment (Chasky-H) [inline client-side/src/app/components/atd-config/atd-config.component.ts:136]: `this,`?

Comment (Chasky-H) [reply client-side/src/app/components/atd-config/atd-config.component.ts:136]: Why are you setting inventory type to DoNothing?

Comment (Chasky-H) [reply client-side/src/app/components/atd-config/atd-config.component.ts:136]: You should no this is the InventoryFIeld is set to None

File: cpi-side/quantity-calculator.ts [hunk 3]
```diff
@@ -1,203 +1,118 @@
...
+                    case ItemAction.Set:
+                        if(value === 0){
+                            return value;
+                        }
+                const min = this.caseBehavior != 'Fix'? this.originalMin: this.getRealMin();
+                //when min > inventory and minx=fix and inv=fix we cannot buy even 1 item
+                if(min > this.normalizedInv && this.invBehavior === 'Fix' && this.minBehavior === 'Fix')
+                {
+                    return 0;
+                }
```
Comment (Chasky-H) [inline cpi-side/quantity-calculator.ts:67]: Is this the right place for this?
In this function?
In this case?

File: cpi-side/quantity-calculator.ts [hunk 4]
```diff
@@ -1,203 +1,118 @@
...
+
+            //value is non negative integer
+            //always fix in case and min
+            //if after the increment by case he is less than real minimum than he should be mean;
+            //if after increment by case he is not divided by case, he should be the next non negative number that divided by case(unless he is bigger than max and inv = fix)
+            getIncrementValue(value: number):QuantityResult {
+                const prevLegalValue = this.fixByCase(value,ItemAction.Decrement) + this.cq;
+                //should return an integer that is no less than value
+                //case there is no interval;
+                if((this.getRealMax() < this.getRealMin()) && this.invBehavior === 'Fix')
```
Comment (Chasky-H) [inline cpi-side/quantity-calculator.ts:99]: Why can't you just check after fixing that the fixed can't be smaller than the existing?

**Review by Chasky-H — CHANGES_REQUESTED** [review]

File: cpi-side/quantity-calculator.ts (hunk 3 above)
Comment (michaelatias) [reply cpi-side/quantity-calculator.ts:67]: I can create a function that handle with edge cases, and that function will invoke before all the fix operations. that function will handle the case you mention in the next note.

File: cpi-side/quantity-calculator.ts (hunk 4 above)
Comment (michaelatias) [reply cpi-side/quantity-calculator.ts:99]: in the fix functions those function are not see the big picture. fix by min - only fix by min, and fix by max only fix buy max. in that case i need to see the whole interval, and realize that in that case i cant do nothing. i can add that case to the edge cases handler.

**Pull request merged**